salt/modules/pkg_resource.py

```diff
@@ -311,8 +311,7 @@ def format_pkg_list(packages, versions_as_list, attr):
     '''
     ret = copy.deepcopy(packages)
     if attr:
-        requested_attr = set(['epoch', 'version', 'release', 'arch',
-                              'install_date', 'install_date_time_t'])
+        requested_attr = {'epoch', 'version', 'release', 'arch', 'install_date', 'install_date_time_t'}
```

Comment:
This needs to stay as a set since we use the `set` type's `&=` operator to modify the set. Changing to a tuple would break this.

Reply:
That one yes. I've missed the part. But other than that, if the set is immutable, I would just prefer tuples instead.